flink-ml-servable-lib/src/main/java/org/apache/flink/ml/classification/logisticregression/LogisticRegressionModelServable.java

```diff
@@ -0,0 +1,148 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/** A Servable which can be used to classifies data in online inference. */
+public class LogisticRegressionModelServable
+        implements ModelServable<LogisticRegressionModelServable>,
+                LogisticRegressionModelParams<LogisticRegressionModelServable> {
+
+    private final Map<Param<?>, Object> paramMap = new HashMap<>();
+
+    private DenseVector coefficient;
+
+    public LogisticRegressionModelServable() {
+        ParamUtils.initializeMapWithDefaultValues(paramMap, this);
+    }
+
+    @Override
+    public DataFrame transform(DataFrame input) {
+        List<Double> predictionResults = new ArrayList<>();
+        List<DenseVector> rawPredictionResults = new ArrayList<>();
+
+        int featuresColIndex = input.getIndex(getFeaturesCol());
+        for (Row row : input.collect()) {
+            Vector features = (Vector) row.get(featuresColIndex);
+            Tuple2<Double, DenseVector> dataPoint = predictOneDataPoint(features, coefficient);
+            predictionResults.add(dataPoint.f0);
+            rawPredictionResults.add(dataPoint.f1);
+        }
+
+        input.addColumn(getPredictionCol(), DataTypes.DOUBLE, predictionResults);
+        input.addColumn(
+                getRawPredictionCol(), DataTypes.VECTOR(BasicType.DOUBLE), rawPredictionResults);
+
+        return input;
+    }
+
+    public LogisticRegressionModelServable setModelData(InputStream... modelDataInputs)
+            throws IOException {
+        Preconditions.checkArgument(modelDataInputs.length == 1);
+
+        DataInputViewStreamWrapper inputViewStreamWrapper =
+                new DataInputViewStreamWrapper(modelDataInputs[0]);
+
+        DenseVectorSerializer serializer = new DenseVectorSerializer();
+        coefficient = serializer.deserialize(inputViewStreamWrapper);
+
+        return this;
+    }
+
+    public static LogisticRegressionModelServable load(String path) throws IOException {
+        LogisticRegressionModelServable servable =
+                ServableReadWriteUtils.loadServableParam(
+                        path, LogisticRegressionModelServable.class);
+
+        try (InputStream fsDataInputStream = ServableReadWriteUtils.loadModelData(path)) {
+            DataInputViewStreamWrapper dataInputViewStreamWrapper =
+                    new DataInputViewStreamWrapper(fsDataInputStream);
+            DenseVectorSerializer serializer = new DenseVectorSerializer();
+            servable.coefficient = serializer.deserialize(dataInputViewStreamWrapper);
+            return servable;
+        }
+    }
+
+    /**
+     * The main logic that predicts one input data point.
+     *
+     * @param feature The input feature.
+     * @param coefficient The model parameters.
+     * @return The prediction label and the raw probabilities.
+     */
+    public static Tuple2<Double, DenseVector> predictOneDataPoint(
```

Comment:
Should we make this method protected so that it won't be used by code outside Flink ML repo to do online prediction?

Would it be better to name this method `Tuple2<Double, DenseVector> transform(Vector feature)` so that it is more consistent with `DataFrame transform(DataFrame input)`?